lib/client.js

```diff
@@ -834,6 +844,12 @@ class Parser {
       }
     }
 
+    /* istanbul ignore next: should be handled by llhttp? */
+    if (request.method !== 'HEAD' && contentLength && bytesRead !== parseInt(contentLength, 10)) {
+      util.destroy(socket, new ResponseContentLengthMismatchError())
+      return -1
+    }
```

Comment:
Why is this ignored by istanbul? Can't we reproduce?

Reply:
I think llhttp doesn't actually allow it, i.e. it shouldn't be possible.

Reply:
I will have to dig into it further. Just too tired right now. Only slept 6 hours in total this week 😞 .

Reply:
Please take some rest 🙏🏻